packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -208,6 +209,10 @@ function useComboboxContent(
   const [selectedOption, setSelectedOption] = useState<ComboboxOption | null>(
     defaultValue || null,
   );
+  useEffect(() => {
+    setSelectedOption(defaultValue || null);
+  }, [selected]);
```

Comment:
Darryl had previously suggested scrapping `const [selectedOption, setSelectedOption]` altogether and instead using `props.selected` directly

this could 100% work for single select, and make it simpler for how to manage what is selected and clearing that selection. `selected` becomes mandatory because it is the state now.

my argument was that multiselect will want that internal state, but obviously we're not building multi select right now so that's not the greatest argument

what are your thoughts on that idea?

Reply:
I tried this out to see how hard it would be. it's surprisingly straightforward

<img width="927" alt="image" src="https://github.com/GetJobber/atlantis/assets/11843215/9d92102a-b7b7-4c05-a41a-4b889511945a">

no more `defaultOption` nor useEffect listening to `props.selected`. the value passed in just is what we use, so if you give us a selection to begin with - cool, that will be selected. if you either give us `null` to begin with or change a value from something to `null` then it's empty. I don't mind it honestly.

is there something you're worried about as far as re rendering on each selection?

Reply:
Just chiming in about rerendering.

If you're talking about the DOM element rerendering. I can assure you that the majority of the time, React (on web) knows not to rerender an element if it didn't drastically change. Even adding a new element inside another element or adding a simple class name won't trigger a full rerender of that element. 

If you use an index as a `key` and the index has changed then it will rerender the items where the index changed.

You can even test this in the DOM. 

![image](https://github.com/GetJobber/atlantis/assets/15986172/3d406e8a-67a5-4e07-b7be-1fadf88d8314)

If you're talking about the functional component itself rerendering all of the function/const within it, no amount of state would stop that from happening since that's how react keeps on reacting to changes. You can use memoization, but that's overkill and putting in resources where it doesn't need to.